annonatate/utils/image.py

```diff
@@ -0,0 +1,77 @@
+            self.tmpfilepath = False
+            self.manifesturl = "{}{}/manifest.json".format(self.origin_url, self.manifestpath)
+            self.createmanifest()
+        else:
+            # handle uploaded image
+            self.file = request_files['file']
+            self.encodedimage = self.file.stream.read()
+
+    def createmanifest(self):
+        try:
+            fac = ManifestFactory()
+            fac.set_base_prezi_uri(self.url)
+            fac.set_base_image_uri(self.imgurl)
+            fac.set_iiif_image_info(2.0, 2)
+            manifest = fac.manifest(ident=self.manifesturl, label=self.request_form['label'])
+            manifest.viewingDirection = self.request_form['direction']
+            manifest.description = self.request_form['description']
+            manifest.set_metadata({"rights": self.request_form['rights']})
+            seq = manifest.sequence()
+            cvs = seq.canvas(ident='info', label=self.request_form['label'])
+            anno = cvs.annotation(ident="{}{}/annotation/1.json".format(self.origin_url, self.manifestpath))
+            img = anno.image(self.iiiffolder, iiif=True)
+        except Exception as e:
+            return {'error': e}
+        if self.tmpfilepath:
+            img.set_hw_from_file(self.tmpfilepath)
+        else:
+            try:
+                img.set_hw_from_iiif()
+            except:
+                try:
+                    response = requests.get("{}{}/info.json".format(self.imgurl, self.iiiffolder))
+                    if response.status_code > 299:
+                        response = requests.get("{}{}".format(self.imgurl, self.iiiffolder))
+                    try:
+                        content = response.json()
+                    except:
+                        return {'error': 'No IIIF image exists at {}{}'.format(self.imgurl, self.iiiffolder)}
+                        # im = Image.open(BytesIO(response.content))
+                        # print(im.size)
+                        # content = {'height': im.size[1], 'width': im.size[0]}
+                    img.height = content['height']
+                    img.width = content['width']
+                except:
+                    return {'error': 'Unable to get height/width for image located at {}{}'.format(self.imgurl, self.iiiffolder)}
+        cvs.height = img.height
+        cvs.width = img.width
+        self.manifest = manifest
+        if type(self.manifest) == dict: # manifest creation failed
+            self.template_name = 'upload.html'
+            self.template_vars = {'error': self.manifest['error']}
+        else:
+            # pb: previously self.manifest was overwritten with the string version, but for testing we need the json version
+            manifest_markdown = self.manifest.toString(compact=False).replace('canvas/info.json', 'info.json').replace('https://{{site.url}}', '{{site.url}}')
+            self.manifest_markdown = "---\n---\n{}".format(manifest_markdown)
```

Comment:
Only note I have, if I remove the else statement is that going to be a problem. The code would look like this: 

if type(self.manifest) != dict: # manifest creation failed
  manifest_markdown = self.manifest.toString(compact=False).replace('canvas/info.json', 'info.json').replace('https://{{site.url}}', '{{site.url}}')
   self.manifest_markdown = "---\n---\n{}".format(manifest_markdown)

Reply:
The change I made was to leave ```self.manifest``` as a dict, and put the markdown string in ```self.manifest_markdown```, instead of storing the markdown in ```self.manifest`` and thereby losing the dict.  This was necessary to make the dict available to the tests at https://github.com/dnoneill/annonatate-sourcecode/blob/937353fef26f54d0c59946c9840a1eb235ac286d/test/test_image.py#L22-L24 . As long as the tests can see the dict, it's ok. The manifest_markdown could be generated outside the else, but it wouldn't be used.. Does that make sense?

Reply:
I am not quite sure. This bit of code is outdated because I found a bug somewhere else. If you wouldn't mind reviewing https://github.com/dnoneill/annonatate-sourcecode/commit/4fcadac0c37f91412b4bb47ace71867efae14ce9 to see if there are any problems you foresee. I ran the tests and they seemed to run.